src/coord/src/catalog/storage.rs

```diff
@@ -244,6 +244,39 @@ impl Connection {
         res
     }
 
+    pub fn get_catalog_content_version(&mut self) -> Result<usize, Error> {
+        let tx = self.inner.transaction()?;
+        let current_setting: Option<String> = tx
+            .query_row(
+                "SELECT value FROM settings WHERE name = 'catalog_content_version';",
+                params![],
+                |row| row.get(0),
+            )
+            .optional()?;
+        let version = match current_setting {
+            Some(v) => v.parse::<i64>().unwrap() as usize,
```

Comment:
Getting the value as a string and then parsing it as an `i64` is roundabout. You should be able to get the value directly as a `u32`, then `usize::cast_from`.

Reply:
```diff
diff --git a/src/coord/src/catalog/storage.rs b/src/coord/src/catalog/storage.rs
index 9de5d071a..3b4c1bcc8 100644
--- a/src/coord/src/catalog/storage.rs
+++ b/src/coord/src/catalog/storage.rs
@@ -246,7 +246,7 @@ impl Connection {
 
     pub fn get_catalog_content_version(&mut self) -> Result<usize, Error> {
         let tx = self.inner.transaction()?;
-        let current_setting: Option<String> = tx
+        let current_setting: Option<u32> = tx
             .query_row(
                 "SELECT value FROM settings WHERE name = 'catalog_content_version';",
                 params![],
@@ -254,7 +254,7 @@ impl Connection {
             )
             .optional()?;
         let version = match current_setting {
-            Some(v) => v.parse::<i64>().unwrap() as usize,
+            Some(v) => v,
             None => {
                 tx.execute(
                     "INSERT INTO settings (name, value) VALUES ('catalog_content_version', 0);",
@@ -264,7 +264,7 @@ impl Connection {
             }
         };
         tx.commit()?;
-        Ok(version as usize)
+        Ok(usize::cast_from(version))
     }
 
     pub fn set_catalog_content_version(&mut self, new_version: usize) -> Result<(), Error> {
```